### PR TITLE
Switch to using Msg instead of JSON

### PR DESCRIPTION
**Pull request by raaidbroad**
I think I've finished addressing everything in the ticket, as well as downstream impacts of said changes. We likely need more test coverage (for example, the `writeToDisk` method is not tested and that concerns me), but that can be added in another PR.

### CODE REVIEW

**Comment by benjamincarlin** [comment]
Add params to your scala docs

**Comment by benjamincarlin** [comment]
Try adding test for when an empty tsv/input is needed for removing, renaming etc

File: v2f/src/test/scala/org/broadinstitute/monster/etl/v2f/V2FUtilsSpec.scala [hunk 1]
```diff
@@ -78,6 +78,52 @@ class V2FUtilsSpec extends PipelineSpec with Matchers {
     )
   }
 
+  // getReadableFiles
+  it should "get TSV files as ReadableFiles given a pattern match" in {
```
Comment (benjamincarlin) [inline v2f/src/test/scala/org/broadinstitute/monster/etl/v2f/V2FUtilsSpec.scala]: add a nested subdirectory such that you can do path/to/files/*/*.txt as well

File: common/src/main/scala/org/broadinstitute/monster/etl/MsgTransformations.scala [hunk 2]
```diff
@@ -44,17 +44,61 @@ object MsgTransformations {
...
+    * Keys not included in the fields input are left out.
+    *
+    * If none of the fields to extract are present in the message,
+    * throw an exception.
+    */
+  def extractFields(fields: Set[String])(msg: Msg): Msg = {
+    val toRet = Obj()
+    val underlying = toRet.obj
+    fields.foreach { field =>
+      if (msg.obj.keySet.contains(Str(field))) {
```
Comment (danxmoran) [inline common/src/main/scala/org/broadinstitute/monster/etl/MsgTransformations.scala]: This `contains` isn't needed. `msg.obj.get(Str(field))` will return `None` if the key isn't present, and the chained `foreach` call will no-op.

File: common/src/main/scala/org/broadinstitute/monster/etl/MsgTransformations.scala [hunk 3]
```diff
@@ -44,17 +44,61 @@ object MsgTransformations {
...
+        msg.obj.get(Str(field)).foreach { value =>
+          underlying.put(Str(field), value)
+        }
+      }
+    }
+    if (toRet.obj.isEmpty) {
+      throw new Exception(
+        s"Failed to extract fields, none of the fields to extract are present"
```
Comment (danxmoran) [inline common/src/main/scala/org/broadinstitute/monster/etl/MsgTransformations.scala]:
```suggestion
        s"Failed to extract fields, none of the fields to extract are present: $toRet"
```
For better debugging

File: common/src/main/scala/org/broadinstitute/monster/etl/MsgTransformations.scala [hunk 4]
```diff
@@ -44,17 +44,61 @@ object MsgTransformations {
...
     * Values in the collected array will appear in the same order as their
     * corresponding field names in the input to this method. Keys not included
-    * in the collection list are left in the message as-is.
+    * in the collection Set are left in the message as-is.
     *
     * If a field-to-collect is missing from the message, the transformation
     * continues (as opposed to throwing an exception).
     */
-  def collectFields(fields: List[String], collectedName: String)(msg: Msg): Msg = {
+  def collectFields(fields: Set[String], collectedName: String)(msg: Msg): Msg = {
```
Comment (danxmoran) [inline common/src/main/scala/org/broadinstitute/monster/etl/MsgTransformations.scala]: Ah, it's important for this one to stay a `List`. Sets are unordered, and we want to make sure that these accumulation functions preserve input order for cases when it's important.

File: common/src/main/scala/org/broadinstitute/monster/etl/MsgTransformations.scala [hunk 5]
```diff
@@ -69,18 +113,18 @@ object MsgTransformations {
...
     * as their corresponding field names in the input to this method. Keys
-    * not included in the concatenation list are left in the message as-is.
+    * not included in the concatenation Set are left in the message as-is.
     *
     * If a field-to-concat is missing from the message, the transformation
     * will throw an exception.
     */
   def concatFields(
-    fields: List[String],
+    fields: Set[String],
```
Comment (danxmoran) [inline common/src/main/scala/org/broadinstitute/monster/etl/MsgTransformations.scala]: This also needs to stay a `List`

File: common/src/test/scala/org/broadinstitute/monster/etl/MsgTransformationsSpec.scala [hunk 6]
```diff
@@ -60,6 +60,84 @@ class MsgTransformationsSpec extends FlatSpec with Matchers {
...
+  // removeFields
+  it should "remove fields in object messages" in {
+    val input = Obj(
+      Str("foo") -> Str("bar"),
+      Str("foobar") -> Int32(123),
+      Str("baz") -> Arr(Str("qux"), Float64(1.23))
+    )
+
+    val removed =
+      MsgTransformations.removeFields(Set("foo"))(input)
```
Comment (danxmoran) [inline common/src/test/scala/org/broadinstitute/monster/etl/MsgTransformationsSpec.scala]: Can you test with two fields? Just to be sure the looping is actually happening.

File: common/src/test/scala/org/broadinstitute/monster/etl/MsgTransformationsSpec.scala [hunk 7]
```diff
@@ -60,6 +60,84 @@ class MsgTransformationsSpec extends FlatSpec with Matchers {
...
+  // extractFields
+  it should "keep fields in object messages" in {
+    val input = Obj(
+      Str("foo") -> Str("bar"),
+      Str("foobar") -> Int32(123),
+      Str("baz") -> Arr(Str("qux"), Float64(1.23))
+    )
+
+    val extracted =
+      MsgTransformations.extractFields(Set("foo"))(input)
```
Comment (danxmoran) [inline common/src/test/scala/org/broadinstitute/monster/etl/MsgTransformationsSpec.scala]: Same, can you keep 2/3?

File: v2f/src/main/scala/org/broadinstitute/monster/etl/v2f/ExtractionPipeline.scala [hunk 8]
```diff
@@ -7,14 +7,15 @@ import com.spotify.scio.values.SCollection
...
 import org.broadinstitute.monster.etl._
-import com.spotify.scio.extra.json._
+import upack.Msg
 
 /**
   * ETL workflow for converting and transforming TSVs from V2F.
   */
 object ExtractionPipeline {
 
   implicit val jsonCoder: Coder[JsonObject] = Coder.kryo[JsonObject]
```
Comment (danxmoran) [inline v2f/src/main/scala/org/broadinstitute/monster/etl/v2f/ExtractionPipeline.scala]: This one can go away now

File: v2f/src/main/scala/org/broadinstitute/monster/etl/v2f/ExtractionPipeline.scala [hunk 9]
```diff
@@ -33,140 +34,149 @@ object ExtractionPipeline {
...
+    val frequencyAnalysisMsgAndFilePaths = V2FExtractionsAndTransforms.extractAndConvert(
       FrequencyAnalysis,
       pipelineContext,
       inputDir = parsedArgs.inputDir,
       relativeFilePath = "*/*.csv"
     )
 
-    val frequencyAnalysisTransformedJsonAndFilePaths = V2FExtractionsAndTransforms
-      .transform(frequencyAnalysisJsonAndFilePaths, FrequencyAnalysis)
+    val frequencyAnalysisTransformedMsgAndFilePaths = V2FExtractionsAndTransforms
```
Comment (danxmoran) [inline v2f/src/main/scala/org/broadinstitute/monster/etl/v2f/ExtractionPipeline.scala]: These `val` names are long to the point of being unreadable, can we simplify?

A useful technique can sometimes be to set up local blocks:
```scala
val frequencyAnalysisOutput = {
  val filePaths = ???
  val rawMessages = doThing(filePaths, ???)
  // etc.
}
val otherTableOutput = {
  val filePaths = ???
  val rawMessages = doThing(filePaths, ???)
  // etc.
}
```
Also, if there's a sequence of SCollection transforms that get applied in order for each table, we could pull those into a helper method.

File: v2f/src/main/scala/org/broadinstitute/monster/etl/v2f/V2FUtils.scala [hunk 10]
```diff
@@ -100,44 +77,25 @@ object V2FUtils {
...
+  ): SCollection[(String, Msg)] => SCollection[(String, Msg)] = {
+    _.transform(s"Adding the ancestry ID from $tableName's tsv path to its Msg object") {
       collection =>
         collection.map {
-          case (filePath, jsonObj) =>
-            val ancestryID = Json.fromString(
+          case (filePath, msgObj) =>
+            val toRet = upack.copy(msgObj)
+            val underlying = toRet.obj
+            val ancestryID = Str(
```
Comment (danxmoran) [inline v2f/src/main/scala/org/broadinstitute/monster/etl/v2f/V2FUtils.scala]: Nitpick: I think it'd be better style to have the `Str` wrapper around `ancestryID` in the call to `update` below, instead of here. That shows more clearly that the `Str` is only there to make the update type-check.

File: v2f/src/main/scala/org/broadinstitute/monster/etl/v2f/ExtractionPipeline.scala (hunk 9 above)
Comment (raaidbroad) [reply v2f/src/main/scala/org/broadinstitute/monster/etl/v2f/ExtractionPipeline.scala]: Definitely in favor of shortening the variable names. I just tried to create a helper function, but some of the tables need to spit out the extracted and converted (but not yet transformed) data to the variant extraction step. Sooo that seems way messier and at that point I think it might make more sense to just keep the structure as is but rename things and maybe scoot some code around.

File: common/src/main/scala/org/broadinstitute/monster/etl/MsgTransformations.scala [hunk 11]
```diff
@@ -44,15 +47,67 @@ object MsgTransformations {
...
+    val toRet = Obj()
+    val underlying = toRet.obj
+    fields.foreach { field =>
+      msg.obj.get(Str(field)).foreach { value =>
+        underlying.put(Str(field), value)
+      }
+    }
+    if (toRet.obj.isEmpty) {
+      throw new Exception(
+        s"Failed to extract fields, none of the fields to extract are present: $toRet"
```
Comment (benjamincarlin) [inline common/src/main/scala/org/broadinstitute/monster/etl/MsgTransformations.scala]: Specify which fields might be helpful for debugging
